dev/com.ibm.ws.cdi.weld/src/com/ibm/ws/cdi/impl/weld/WebSphereCDIDeploymentImpl.java

```diff
@@ -718,4 +720,27 @@ public CDI<Object> getCDI() {
         return this.cdi;
     }
 
+    /** {@inheritDoc} */
+    @Override
+    public Collection<URL> getUnversionedBeansXmlURLs() {
+        Collection<URL> unversionedBeansXmlURLs = new ArrayList<URL>();
+        for (WebSphereBeanDeploymentArchive bda : getWebSphereBeanDeploymentArchives()) {
+            BeansXml beansXml = bda.getBeansXml();
+            if (beansXml != null) { //check that there is a beans.xml file
+                URL parsedURL = beansXml.getUrl();
+                //if the URL is null then that means it was an empty beans.xml file
+                //note that this may be an undocumented "feature" of the Weld SPI
+                if (parsedURL != null) {
+                    //if the beans.xml was not an empty file then check if the version was set or not
+                    boolean unversionedBeansXml = beansXml.getVersion() == null;
+                    if (unversionedBeansXml) {
+                        URL unversionedBeansXmlURL = bda.getBeansXmlResourceURL();
+                        unversionedBeansXmlURLs.add(unversionedBeansXmlURL);
+                    }
+                }
+            }
+        }
```

Comment:
i disagree ... having multiple if blocks is easier to read, particularly given that I can comment next to each one. Streams are definitely harder to read and perform worse so why use them unless really needed.

Reply:
Right, but I'd suggest something like this to avoid so many nested blocks:

```suggestion
        for (WebSphereBeanDeploymentArchive bda : getWebSphereBeanDeploymentArchives()) {
            BeansXml beansXml = bda.getBeansXml();
            if (beansXml == null) { //check that there is a beans.xml file
                continue;
            }
            
            URL parsedURL = beansXml.getUrl();
            //if the URL is null then that means it was an empty beans.xml file
            //note that this may be an undocumented "feature" of the Weld SPI
            if (parsedURL == null) {
                continue;
            }
            
            //if the beans.xml was not an empty file then check if the version was set or not
            boolean unversionedBeansXml = beansXml.getVersion() == null;
            if (unversionedBeansXml) {
                URL unversionedBeansXmlURL = bda.getBeansXmlResourceURL();
                unversionedBeansXmlURLs.add(unversionedBeansXmlURL);
            }
        }
```

I'd also suggest than when it's laid out like this, the `parsedURL` and `unversionedBeansXml` variables are superfluous.

Reply:
ok